pie/utils/objects.py

```diff
@@ -96,3 +96,84 @@ def check(reaction, user):
                     with contextlib.suppress(nextcord.Forbidden):
                         await message.remove_reaction("▶️", user)
                     await message.edit(embed=self.pages[pagenum])
+
+
+class ConfirmView(nextcord.ui.View):
+    """Class for making confirmation embeds easy.
+    The right way of getting response is first calling wait() on instance,
+    then checking instance attribute `value`.
+    Attributes:
+        value: True if confirmed, False if declined, None if timed out
+        ctx: Context of command
+        message: Confirmation message
+    Args:
+        ctx: The context for translational and sending purposes.
+        embed: Embed to send.
+        timeout: Number of seconds before timeout. `None` if no timeout
+    """
+
+    def __init__(
+        self,
+        ctx: commands.Context,
+        embed: nextcord.Embed,
+        timeout: Union[int, float, None] = 300,
+        delete: bool = False,
+    ):
+        super().__init__(timeout=timeout)
+        self.value: Optional[bool] = None
+        self.ctx = ctx
+        self.embed = embed
+        self.delete = delete
+
+    async def send(self):
+        """Sends message to channel defined by command context.
+        Returns:
+            True if confirmed, False if declined, None if timed out
+        """
+        self.add_item(
+            nextcord.ui.Button(
+                label=_(self.ctx, "Confirm"),
```

Comment:
We'd have to patch PoPie to allow `self.ctx` instead of just `ctx`. As a workaround you can use `ctx = self.ctx` at the begginging of the function, but that seems annoying.

@Czechbol should we support this edge case in PoPie?

Reply:
Solved by sinus-x's update of Popie